src/nvim/tui/tui.c

```diff
@@ -225,6 +225,25 @@ void tui_handle_term_mode(TUIData *tui, TermMode mode, TermModeState state)
   }
 }
 
+/// Query the terminal emulator to see if it supports extended underline.
+///
+/// Try to set an undercurl, followed by a DECRQSS query.
+static void tui_query_extended_underline(TUIData *tui)
+{
+  out(tui, S_LEN("\x1b[0m\x1b[4:3m\x1bP$qm\x1b\\"));
```

Comment:
nit: 

Add a comment on why we are resetting the attribute first so it's not lost with time.